packages/models/src/models/CalendarEvent.ts

```diff
@@ -149,4 +150,98 @@ export class CalendarEventRaw extends BaseRaw<ICalendarEvent> implements ICalend
 			endTime: { $exists: true, $gte: endTime },
 		});
 	}
+
+	public findEventsToScheduleNow(now: Date, endTime: Date): FindCursor<ICalendarEvent> {
+		return this.find(
+			{
+				startTime: { $gte: now, $lt: endTime },
+				busy: { $ne: false },
+				endTime: { $exists: true },
+			},
+			{
+				sort: { startTime: 1 },
+				projection: {
+					_id: 1,
+					uid: 1,
+					startTime: 1,
+					endTime: 1,
+				},
+			},
+		);
+	}
+
+	public async findNextFutureEvent(startTime: Date): Promise<ICalendarEvent | null> {
+		return this.findOne(
+			{
+				startTime: { $gte: startTime },
+				busy: { $ne: false },
+				endTime: { $exists: true },
+			},
+			{
+				sort: { startTime: 1 },
+				projection: {
+					startTime: 1,
+				},
+			},
+		);
+	}
+
+	public findEventsStartingNow(now: Date, offset = 1000): FindCursor<ICalendarEvent> {
```

Comment:
Can we use named parameters, since we are receiving a "magic" number, it would be easier to understand what the number parameter is about. WDYT?